pkg/ping/runtime_service.go

```diff
@@ -21,31 +24,44 @@ import (
 	"github.com/pluralsh/deployment-operator/pkg/client"
 )
 
-func (s *ServiceReconciler) ScrapeKube(ctx context.Context) {
-	logger := log.FromContext(ctx)
-	logger.Info("attempting to collect all runtime services for the cluster")
+const runtimeServicePingerName = "runtime service pinger"
+
+func RunRuntimeServicePingerInBackgroundOrDie(ctx context.Context, pinger *Pinger, duration time.Duration) {
+	klog.Info("starting ", runtimeServicePingerName)
+
+	err := helpers.BackgroundPollUntilContextCancel(ctx, duration, true, true, func(_ context.Context) (done bool, err error) {
```

Comment:
same should be sure to jitter

Reply:
(good default jitter is like 1/2 or 1/3 of the duration

Reply:
done